### PR TITLE
Use jenkins nodes running nodejs 8 - Closes #1001

### LINKED ISSUE #1001: Modify Jenkinsfile to accept parameter that modifies node selection
This is to allow the choice of running a job on either nodejs 6 nodes or nodejs 8 nodes.

### PR DESCRIPTION
**Pull request by ghost**
### What was the problem?

Core >= 1.2 requires node 8 but jenkins nodes were using version 6.

### How did I fix it?

Changed the labels in the various `Jenkinsfile*` files so that jenkins nodes using node 8 get picked.
The `lock` was removed from `Jenkinsfile.network` since no lockable resources were defined in Jenkins anyway.

### How to test it?

Make sure all the jenkins builds pass with those changes.

### Review checklist

* The PR resolves #1001
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

**Review by Nazgolze — APPROVED** [review]

**Review by diego-G — CHANGES_REQUESTED** [review]
Shall we update this as well https://github.com/LiskHQ/lisk/blob/5e8a12b8034038f17e2bf075da3affaa45426065/Jenkinsfile#L378 ?

**Comment by fchavant** [comment]
@diego-G no, this has nothing to do with the label (but is confusing)

**Comment by diego-G** [comment]
@fchavant what about the comments?

**Comment by fchavant** [comment]
@diego-G same think, they need not be changed (we should make them more meaningful but that would be a different PR)

File: Jenkinsfile [hunk 1]
```diff
@@ -151,7 +151,7 @@ def report() {
 	}
 }
 
-lock(resource: "Lisk-Core-Nodes", inversePrecedence: true) {
+lock(resource: "Lisk-Core-Nodes8", inversePrecedence: true) {
```
[the remark this answers is not in the document]
Comment (fchavant) [reply Jenkinsfile]: Sure.

**Review by ManuGowda — APPROVED** [review]
@fchavant what will happen to these labels when we upgrade to node 10? is this change only specific to node 8 for now?

**Comment by fchavant** [comment]
@ManuGowda I guess we'll create new labels then.

**Review by diego-G — APPROVED** [review]

**Review by MaciejBaj — APPROVED** [review]

**Pull request merged**